src/Microsoft.AspNetCore.Sockets.Client.Http/Internal/LoggingHttpMessageHandler.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+namespace Microsoft.AspNetCore.Sockets.Client.Http.Internal
+{
+    public class LoggingHttpMessageHandler : DelegatingHandler
+    {
+        private readonly ILogger<LoggingHttpMessageHandler> _logger;
+
+        public LoggingHttpMessageHandler(HttpMessageHandler inner, ILoggerFactory loggerFactory) : base(inner)
+        {
+            _logger = (loggerFactory ?? NullLoggerFactory.Instance).CreateLogger<LoggingHttpMessageHandler>();
```

Comment:
You don't need to be this defensive here. Just don't accept a null logger factory.